kolibri/plugins/learn/assets/src/views/TopicsPage/index.vue

```diff
@@ -620,56 +617,58 @@
       channelTitle() {
         return this.channel ? this.channel.name : '';
       },
-      resources() {
-        return this.contents.filter(content => content.kind !== ContentNodeKinds.TOPIC);
-      },
       childrenToDisplay() {
         return this.windowBreakpoint === 2 || this.windowBreakpoint > 4 ? 4 : 3;
       },
       gridType() {
         return this.windowBreakpoint > 4 ? 2 : 1;
       },
-      resourcesDisplayed() {
-        // if no folders are shown at this level, show more resources to fill the space
-        // or if the user has explicitly requested to show more resources
-        if (!this.topics.length || this.showMoreResources) {
-          return this.resources;
-        }
-        return this.resources.slice(0, this.childrenToDisplay);
-      },
-      moreResources() {
-        return this.resourcesDisplayed.length < this.resources.length;
-      },
       topics() {
         return this.contents
           .filter(content => content.kind === ContentNodeKinds.TOPIC)
-          .filter(t => t.children && t.children.results.length)
-          .map(t => {
-            let topicChildren = t.children.results;
+          .filter(t => t.children && t.children.results.length);
+      },
+      contentsForDisplay() {
+        return this.contents
+          .filter(
+            c =>
+              c.kind !== ContentNodeKinds.TOPIC ||
+              ((this.subTopicId ? c.id === this.subTopicId : true) &&
```

Comment:
is this to "disqualify" content where the id of the content is not equal to the current subtopic id?

Reply:
Yes, this is indeed the case, fixing.

Reply:
Amusingly, this bug is also extant on release-v0.17.x - so another thing fixed!